packages/decision-tree/src/DecisionTree.tsx

```diff
@@ -2,17 +2,17 @@ import React, { FC } from 'react';
 import { StepBar } from './components/StepBar';
 import { CallToActions } from './components/CallToActions';
 import { Question } from './components/Question';
+import { DecisionTreeProps, Steps } from './types';
 import { usePagination, useHandleAnswers, PaginationActions, HandleAnswersActions } from './hooks';
-import { DecisionTreeProps, DecisionTreeStaticMembers, Steps } from './types';
 import { Flex } from '@cmpsr/components';
-import { CallToActionsProps } from '@components/CallToActions/types';
-import { StepBarProps } from '@components/StepBar/types';
+import { normalizeQuestionnaire } from './DecisionTree.normalizer';
 
-export const DecisionTree: FC<DecisionTreeProps> & DecisionTreeStaticMembers = ({ questionnaire, callback }) => {
-  const steps: Steps = questionnaire.sections.map(({ id, name }) => ({ id, name }));
-  const initialState = {
-    currentQuestion: questionnaire.nextQuestion.questionId,
-    currentSection: questionnaire.nextQuestion.sectionId,
+export const DecisionTree: FC<DecisionTreeProps> = ({ questionnaire, callback, firstQuestion, renderSectionIntro }) => {
+  const normalizedQuestionnaire = normalizeQuestionnaire(questionnaire);
```

Comment:
question: is it worthy to use `useMemo` in here?

Reply:
maybe, do you mean the normalize function, the whole component? maybe the question component would benefit too.